src/Chapter3.hs

```diff
@@ -480,6 +520,50 @@ After defining the city, implement the following functions:
    and at least 10 living __people__ inside in all houses of the city totally.
 -}
 
+data Castle = Castle 
+  { castleName :: String
+  , hasWalls :: Bool
+  } deriving (Show)
+
+makeCastle :: String -> Bool -> Castle
+makeCastle n withWalls = Castle { castleName = n, hasWalls = True }
+
+data PersonCount = One | Two | Three | Four deriving (Show)
+data House = House PersonCount deriving (Show)
```

Comment:
Now, after you learned about newtypes, you can reuse it here as well :slightly_smiling_face: 
```suggestion
newtype House = House PersonCount deriving (Show)
```